### PR TITLE
Do not install configure_file output if install_dir is empty. Closes …

### PR DESCRIPTION
**Pull request by jpakkane**
…#3270.

### CODE REVIEW

File: docs/markdown/Reference-manual.md [hunk 1]
```diff
@@ -173,7 +173,8 @@ These are all the supported keyword arguments:
   mode, all the variables in the `configuration:` object (see above)
   are written to the `output:` file.
 - `install_dir` the subdirectory to install the generated file to
-  (e.g. `share/myproject`), if omitted the file is not installed.
+  (e.g. `share/myproject`), if omitted or given the value of empty
+  string,the file is not installed.
```
Comment (sarum9in) [inline docs/markdown/Reference-manual.md]: Please add space after the comma.

File: mesonbuild/interpreter.py [hunk 2]
```diff
@@ -2973,9 +2973,11 @@ def func_configure_file(self, node, args, kwargs):
             conffile = os.path.normpath(inputfile.relative_name())
             if conffile not in self.build_def_files:
                 self.build_def_files.append(conffile)
-        # Install file if requested
+        # Install file if requested, we check for the empty string
+        # for backwards compatibility. That was the behaviour before
+        # 0.45.1 so preserve it.
         idir = kwargs.get('install_dir', None)
-        if isinstance(idir, str):
+        if isinstance(idir, str) and idir != '':
```
Comment (sarum9in) [inline mesonbuild/interpreter.py]: I am not sure if this is used throughout meson codebase but [PEP8](https://www.python.org/dev/peps/pep-0008/#id51) recommends `if isinstance(idir, str) and idir:`

**Comment by jpakkane** [comment]
Fixed.

File: test cases/common/16 configure file/meson.build [hunk 3]
```diff
@@ -131,3 +131,9 @@ configure_file(
   configuration : conf6
 )
 test('test6', executable('prog6', 'prog6.c'))
+
+# test empty install dir string
+cfile = configure_file(input : 'config.h.in',
+output : 'do_not_get_installed.h',
```
Comment (sarum9in) [inline test cases/common/16 configure file/meson.build]: Indentation inconsistency.

**Pull request merged**